src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,95 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int INITIAL_SIZE = 16;
+    private static final int INCREASE_ARRAY_SIZE_INDEX = 2;
+
+    private int size;
+    private int threshold;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = new Node[INITIAL_SIZE];
+        this.threshold = (int)(table.length * LOAD_FACTOR);
+    }
+
+    public static class Node<K, V> {
+        private K key;
+        private V value;
+        private Node<K, V> nextNode;
+
+        public Node(K key, V value, Node<K, V> nextNode) {
+            this.key = key;
+            this.value = value;
+            this.nextNode = nextNode;
+        }
+    }
 
     @Override
     public void put(K key, V value) {
+        if (size > threshold) {
+            resize();
+        }
 
+        int index = hashKey(key);
+        Node<K, V> addNode = table[index];
+        if (addNode == null) {
+            table[index] = new Node<K,V>(key,value,null);
+            size++;
+            return;
+        }
```

Comment:
let's remove `if` block and move all logic from its body to the end of the method

Reply:
done